pkgs/development/libraries/frozen/default.nix

```
@@ -0,0 +1,62 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, meson
+, ninja
+, writeText
+}:
+
+stdenv.mkDerivation rec {
+  pname = "frozen";
+  # pin to a newer release, if frozen releases again, see cesanta/frozen#72
+  version = "unstable-2021-02-23";
+
+  src = fetchFromGitHub {
+    owner = "cesanta";
+    repo = "frozen";
+    rev = "21f051e3abc2240d9a25b2add6629b38e963e102";
+    sha256 = "sha256-BpuYK9fbWSpeF8iPT8ImrV3CKKaA5RQ2W0ZQ03TciR0=";
+  };
+
+  nativeBuildInputs = [ meson ninja ];
+
+  # frozen only has internal make and bazel support
+  # as it only consists of two source files, adding a
+  # small meson file on the fly is the easiest option for
+  # cross compilation support
+  #
+  # delete if cesanta/frozen#71 gets merged
+  mesonBuildFile = writeText "meson.build" ''
+    project(
+        'frozen',
+        'c',
+        default_options: [
+            'c_args=-Wextra -fno-builtin -pedantic',
+            'c_std=c99',
+            'werror=true'
+        ],
+        license: 'Apache-2.0',
+        version: '20210223'
+    )
+
+    library(
+        'frozen',
+        'frozen.c',
+        install: true
+    )
+
+    install_headers('frozen.h')
+  '';
+
+  preConfigure = ''
+    cp $mesonBuildFile meson.build
```

After creating the meson.build file,

```suggestion
    cp ${meson.build} meson.build
```

---

Thanks for the hint, I had to use ${./meson.build}.